src/core/src/CubicSpline.cpp

```diff
@@ -0,0 +1,175 @@
+/*!
+ * @file CubicSpline.cpp
+ * @author Stefano Dafarra
+ * @copyright 2016 iCub Facility - Istituto Italiano di Tecnologia
+ *            Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ * @date 2017
+ *
+ */
+
+#include <iDynTree/Core/CubicSpline.h>
+#include <iDynTree/Core/EigenHelpers.h>
+#include <Eigen/Dense>
+#include <iostream>
+#include <cmath>
+
+
+iDynTree::CubicSpline::CubicSpline()
+:m_v0(0)
+,m_vf(0)
+,m_a0(0)
+,m_af(0)
+{
+    m_coefficients.clear();
+    m_velocities.resize(0);
+    m_time.resize(0);
+    m_y.resize(0);
+    m_T.resize(0);;
+}
+
+iDynTree::CubicSpline::~CubicSpline()
+{}
+
+
+
+bool iDynTree::CubicSpline::setData(const iDynTree::VectorDynSize& time, const iDynTree::VectorDynSize& yData)
+{
+    if((time.size() == 0) && (yData.size() == 0)){
+        std::cerr << "[CUBICSPLINE] The input data are empty!" << std::endl;
```

Comment:
If you perpend the error messages with `[ERROR]`, they will be nicely formatted by the yarplogger.